packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http-private.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  HttpConnectionType,
+  HttpIntegrationType,
+  HttpRouteIntegrationBindOptions,
+  HttpRouteIntegrationConfig,
+  IHttpRouteIntegration,
+  PayloadFormatVersion,
+  HttpMethod,
+  IVpcLink,
+} from '@aws-cdk/aws-apigatewayv2';
+import * as ec2 from '@aws-cdk/aws-ec2';
+
+
+/**
+ * Properties to initialize a new `HttpProxyPrivateIntegration`.
+ */
+export interface HttpProxyPrivateIntegrationProps {
+  /**
+   * The ARN of the target resource.
+   * One of ApplicationLoadBalancerListener, NetworkLoadBalancerListener or CloudMapDiscoveryService
+   */
+  readonly targetArn: string;
+
+  /**
+   * The vpc link to be used for the private integration
+   *
+   * @default - a new VpcLink is created
+   */
+  readonly vpcLink?: IVpcLink;
+
+  /**
+   * The vpc for which the VpcLink needs to be created
+   *
+   * @default - undefined
```

Comment:
What does it mean for this to be left undefined? When would a customer not specify this?

Reply:
The `vpc` and `vpcLink` props are both optional with at least one of them needs to be present in the props (we create a `vpcLink` if it is not provided but vpc is).

Reply:
Ok, makes sense.